### DISCUSSION TITLE
Start-Up resources Built-In to the Engine

### DISCUSSION THREAD

**Discussion by ahmutkadim** [godotengine/godot-proposals#7269] [General Discussions]
A feature that I saw in some game engines is not present on Godot. This feature is very easy to implement and saves a lot of time which is why I am mentioning it.

When we wish to make a demo of our game, we would like to use simple, primitive graphics, like squares, triangles and circles. However these primitive resources are not present in Godot editor so we have to import them each time we need them, from an outside source. This can take time for especially in prototype phase.

What I ask is to implement this primitive resources like primitive sprite graphics and simple 3D materials directly to Godot Engine. For instance, when you view the "texture" property of a sprite2D node in the editor, there can be an option of "new square". And we can modify its color with modulation assuming that default square will have a color of white.

This would be great if implemented in core editor of the engine since such primitive resources can be used for many things, from placeholders to debug.

**Comment by theraot** [comment]
Try `Polygon2D` instead of `Sprite2D` for your triangle or other, well, polygons. *I won't be against tooling in core to quickly make regular polygons and stars (there are addons to that end already).*

About the square texture, consider using a `GradientTexture2D`, by default the gradient has two color ends, remove one if you want a solid color, and you can modify the remaining one to whatever you color you want... So you don't need to resource to modulation.

*You can also make the `GradientTexture2D` make a circle if you set the fill to radial, and use alpha.*

---

You also mention simple 3D materials... Is there something lacking from the `StandardMaterial3D`? you can create a new one, set the albedo color, and call it done.

**Comment by ahmutkadim** [comment]
Firstly I must apologize for my claim about 3D, I mostly worked on 2D and am trying to start at 3D, I should not given an example something I have not fully comprehended yet, I am sorry. Secondly, thanks so much for the suggestions about how I can make polygons and primitive shapes in Godot with different resources. I will probably use them in my next project if I remember. But I totally do not claim that it is certainly impossible to generate such shapes on Godot, I think the problem is ease of use and clarity. You know how to generate a square or circle with GradientTexture2D because you have experience with the engine and know enough resources to create one. However it might not be too obvious for a beginner. Was not clear for me in  the first time since I have not needed to learn GradientTexture2D before. Also some of the suggestions you suggested involve multiple steps which makes them as slow as making a shape in a vector image editor. Please do not misunderstand, I really liked your alternatives and they are really useful if what I propose is not added (probably will not). However I think having a separate resource for some of the primitives can be better, plus how much it would cost to add such thing the core.

**Comment by dalexeev** [comment]
* Related: #5294.
